Userland/Applications/Calculator/KeypadValue.h

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright (c) 2021, the SerenityOS developers.
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <AK/Math.h>
+#include <AK/String.h>
+#include <AK/Types.h>
+
+class KeypadValue {
+    friend class Keypad;
+
+public:
+    KeypadValue(i64, u8);
+    KeypadValue(i64);
+
+    //helper lambda for KeypadValue's operators should have access to its private members
+    //because we have default template arguments, we need to define the lambda inside the class declaration
+    template<typename T, typename F = T(KeypadValue const&, KeypadValue const&, i64, i64, bool)>
```

Comment:
The default doesn't really do anything here, the type is deduced from the argument directly.